ducktape/services/service.py

```diff
@@ -51,11 +52,17 @@ class Service(TemplateRenderer):
     # }
     logs = {}
 
-    def __init__(self, context, num_nodes, *args, **kwargs):
+    # TODO: I changed this from num_nodes to nodes but I may not have updated all places where this constructor is called.
```

Comment:
It doesn't look critical (we use a named parameter for `num_nodes` to the `@cluster` annotation, but I haven't found it in the implementation of services). But let's avoid the breaking change since we caught it. I was actually thinking of this solution:

```
def __init__(self, context, num_nodes=None, nodes=None, *args, **kwargs):
```

and then checking that one of `num_nodes` or `nodes` is not `None`. I think that keeps compatibility but lets people use `nodes` as a named parameter for the new node spec.